examples/p4est_2d_dgsem/elixir_advection_meshview.jl

```diff
@@ -0,0 +1,66 @@
+using OrdinaryDiffEqSSPRK, OrdinaryDiffEqLowStorageRK
+using Trixi
+
+###############################################################################
+# Most basic p4est mesh view setup where the entire domain
+# is part of the single mesh view.
+
+advection_velocity = (0.2, -0.7)
+equations = LinearScalarAdvectionEquation2D(advection_velocity)
+
+# Create DG solver with polynomial degree = 3 and (local) Lax-Friedrichs/Rusanov flux as surface flux
+solver = DGSEM(polydeg = 3, surface_flux = flux_lax_friedrichs)
+
+coordinates_min = (-1.0, -1.0) # minimum coordinates (min(x), min(y))
+coordinates_max = (1.0, 1.0) # maximum coordinates (max(x), max(y))
+
+trees_per_dimension = (8, 8)
+
+# Create parent P4estMesh with 8 x 8 trees and 8 x 8 elements
+parent_mesh = P4estMesh(trees_per_dimension, polydeg = 3,
+                        coordinates_min = coordinates_min,
+                        coordinates_max = coordinates_max,
+                        initial_refinement_level = 0)
+
+# Define the mesh view covering the whole parent mesh.
+cell_ids = Vector(1:Trixi.ncells(parent_mesh))
+mesh = P4estMeshView(parent_mesh, cell_ids)
+
+# A semidiscretization collects data structures and functions for the spatial discretization
+semi = SemidiscretizationHyperbolic(mesh, equations, initial_condition_convergence_test,
+                                    solver)
+
+###############################################################################
+# ODE solvers, callbacks etc.
+
+# Create ODE problem with time span from 0.0 to 1.0
+ode = semidiscretize(semi, (0.0, 1.0))
+
+# At the beginning of the main loop, the SummaryCallback prints a summary of the simulation setup
+# and resets the timers
+summary_callback = SummaryCallback()
+
+# The AnalysisCallback allows to analyse the solution in regular intervals and prints the results
+analysis_callback = AnalysisCallback(semi, interval = 100)
+
+# The SaveSolutionCallback allows to save the solution to a file in regular intervals
+save_solution = SaveSolutionCallback(interval = 100,
+                                     solution_variables = cons2prim)
+
+# The StepsizeCallback handles the re-calculation of the maximum Δt after each time step
+stepsize_callback = StepsizeCallback(cfl = 1.6)
+
+# Create a CallbackSet to collect all callbacks such that they can be passed to the ODE solver
+callbacks = CallbackSet(summary_callback, save_solution, stepsize_callback)
```

Comment:
```suggestion
callbacks = CallbackSet(summary_callback, analysis_callback, save_solution,
                        stepsize_callback)
```
The `analysis_callback` was unused before.

Reply:
The 'analysis_callback' is not yet implemented for pest mesh views.

Reply:
How difficult would it be to make the `P4estMeshView` work with the `analysis_callback`? The `analysis_callback` is quite fundamental and also used in the tests to determine the `l2` and `linf` error (tests are currently failing because the `analysis_callback` is not present). Therefore, it would be nice to be able to use it.

Reply:
As with StructuredMeshView, this is meant to be used in coupled simulations. Once they are implemented, the analysis callback would be on the coupled system.

Reply:
I see, but then we still need to find a way to fix [the failing test](https://github.com/trixi-framework/Trixi.jl/actions/runs/13840485073/job/38726463047?pr=2110#step:7:2459).

Reply:
Yeah, I just saw that. But this fail is new, so I should be able to fix it by going through the commit history.

Reply:
I think I see. Previously, you defined the `analysis_callback` in the elixir, but didn't use it in the `CallbackSet`. This should fix it.

Reply:
But please add a comment in the elixir why you don't use the `analysis_callback` in the elixir, but still define it. Otherwise this is quite confusing.

Reply:
Just done that. Thanks.